tests/unit/distribution/test_library_client_initialization.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+"""
+Unit tests for LlamaStackAsLibraryClient initialization error handling.
+
+These tests ensure that users get proper error messages when they forget to call
+initialize() on the library client, preventing AttributeError regressions.
+"""
+
+import pytest
+
+from llama_stack.distribution.library_client import (
+    AsyncLlamaStackAsLibraryClient,
+    LlamaStackAsLibraryClient,
+)
+
+
+class TestLlamaStackAsLibraryClientInitialization:
```

Comment:
some of these tests are repetitive and not necessary

Reply:
@mattf no I am not concerned explicitly but I have a very strong bias towards _fewer_ but higher signal tests. Each test is a clear maintenance burden (and often over time leads to people just mashing up random mocks to get them to pass without actually providing useful coverage). If you see some of the parameterizations for a couple tests at the bottom, they seem unnecessary.

Reply:
@ashwinb want me to strike the proviers.list? it's mostly the same as models.list. the chat.completions test two different paths given the stream/non-stream. the sync/async share code paths, but i included them both because the impl could change.

Reply:
yeah sure. feel free to review and do what you feel is necessary and then just land it.